src/daemon/rpmostreed-transaction-types.cxx

```diff
@@ -601,6 +601,10 @@ typedef struct {
   char  **install_pkgs; /* strv but strings owned by modifiers */
   GUnixFDList *install_local_pkgs;
   char  **uninstall_pkgs; /* strv but strings owned by modifiers */
+  char  **enable_modules; /* strv but strings owned by modifiers */
```

Comment:
Unrelated but I think this PR demonstrates really well we still have *far* too much "mechanical stuff" to plumb through to add new stuff like this.  So many conversions from GVariant to C data types (`char**`) back into `GKeyFile` then back into `char**` etc...with Rust and C++ in there too...

Hmm.  You know what's tempting is to change `UpdateDeployment` to go from `GVariant a{sv}` -> `Treefile` *internally*.  That would be a huge step towards https://github.com/coreos/rpm-ostree/issues/2326 in letting the API take a new treefile directly.

Reply:
The thing is that `UpdateDeployment` is "delta-based", whereas `Treefile` is about layering. E.g. we need to be able to say "stop layering this package".

Reply:
Yep, good point.  We could at least represent additions via treefile.  But maybe it'd be messier to have a half-declarative and half-imperative API?

Reply:
Random thought: We could have two treefiles: One for additions, and one for removals.  The "removal" treefile would have
```
packages:
  - foo
  - bar
 ```
as usual, but the code would interpret it as things to remove?